synapse/storage/devices.py

```diff
@@ -210,12 +217,15 @@ def _get_device_update_edus_by_remote(self, destination, from_stream_id, query_m
                 destination, user_id, from_stream_id
             )
             for device_id, device in iteritems(user_devices):
-                stream_id = query_map[(user_id, device_id)]
+                stream_id, _ = query_map[(user_id, device_id)]
                 result = {
                     "user_id": user_id,
                     "device_id": device_id,
                     "prev_id": [prev_id] if prev_id else [],
                     "stream_id": stream_id,
+                    "context": query_map[(user_id, device_id)][1]
+                    if opentracing.whitelisted_homeserver(destination)
```

Comment:
do we need to do this filtering, given we also filter it when the rows go into `device_lists_outbound_pokes`?

Reply:
This is indeed redundant